_posts/2021-09-07-dd_00003_logic_gates.md

```diff
@@ -0,0 +1,347 @@
+principle. In the post following that, we will see how we can use transistors to
+build logic gates.
+
+Let's get started!
+
+## Logic Operations and Truth Tables
+
+As the name suggests, logic gates perform so-called **logic operations**. A
+logical operation takes some inputs, and combines them in a specific way to
+produce an output. Each input can be either "true" or "false" (or "1", or "0").
+Same holds for the output. For example, the logical "**AND**" operation has two
+inputs, A and B, and one output, Z. The output Z will only be "true" when both A
+_and_ B are "true", hence "and".
+
+The behavior of any logical operation can be respresented using a special table,
+called the **truth table**. The truth table shows for all possible input
+combinations the corresponding output values, thereby fully characterizing the
+function of the particular logic operation. For example, the truth table of the
+AND operation looks like this:
+
+| A   | B   | Z   |
+| --- | --- | --- |
+| 0   | 0   | 0   |
+| 0   | 1   | 0   |
+| 1   | 0   | 0   |
+| 1   | 1   | 1   |
+
+As you can see, the output, Z, is only "on" (1) when _both_ inputs, A and B, are
+"on"; otherwise, the output stays "off" (0). Let's look at the truth table of
+another logic operation, the **OR**:
+
+| A   | B   | Z   |
+| --- | --- | --- |
+| 0   | 0   | 0   |
+| 0   | 1   | 1   |
+| 1   | 0   | 1   |
+| 1   | 1   | 1   |
+
+Can you see the difference of the OR compared to the AND operation? The OR
+operation produces a "1" at its output, when _at least_ one input is set to "1".
+This behavior is also what its name suggests: A _or_ B (and that also includes
+the case when both A and B are 1). Another interesting logic operation is the
+**NOT** operation:
+
+| A   | Z   |
+| --- | --- |
+| 0   | 1   |
+| 1   | 0   |
+
+The NOT operation only has one input and one output. Again, as the name
+suggests, the NOT operation simply produces the _inverse_ (or the _negation_) of
+its input.
+
+Now, buckle up, your mind is about to be blown away: With only the three logic
+operations we presented you above, namely the AND, OR and the NOT operations, it
+is possible to construct ANY digital circuit, even the processor you are using
+to read this blog post[^1]!
+
+Next, we will learn how this is done in practice.
+
+[^1]:
+    For the people that are already screaming: Yes, you need a bit more than
+    just logic operations to build a processor, but for the digital part of the
+    chip, the statement is true.
+
+## Basic Logic Gates
+
+Logic operations are implemented in hardware using **logic gates**. Just like a
+logic operation, a logic gate has inputs and output(s). For example, this is the
+logic gate for the AND operation (or short: _AND gate_):
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-and-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
```

Comment:
Do you want to tell that there are also other ways how they can look like (ANSI vs IEC for example)

Reply:
Not sure if this fits in this introductory post. Maybe later when we start to cover more complex circuits?

Reply:
Maybe a small note? Just one sentence. "There are other ways.... that are not covered at this point ...." Something like this

Reply:
Like this: bfe4c33 ?